xtask/src/gendocs.rs

```diff
@@ -0,0 +1,94 @@
+//! Convert Phylum CLI to markdown.
+
+use std::path::{Path, PathBuf};
+use std::{env, fs};
+
+use anyhow::Result;
+use clap_markdown::Generator;
+use phylum_cli::app;
+
+/// Output directory.
+const OUTPUT_DIR: &str = "./docs/command_line_tool";
+
+/// Template directory.
+const TEMPLATE_DIR: &str = "./doc_templates";
+
+/// File header inserted at the top of each page.
+const HEADER: &str = "---
+title: {PH-TITLE}
+category: 6255e67693d5200013b1fa3e
+hidden: false
+---";
+
+/// Generate Phylum CLI documentation.
+pub fn gendocs() -> Result<()> {
+    // Create target directory.
+    fs::create_dir_all(Path::new(OUTPUT_DIR))?;
+
+    // Store all generated markdown pages in target directory.
+    for (path, markdown) in pages()? {
+        fs::write(path, markdown)?;
+    }
+
+    Ok(())
+}
+
+/// Get a vec with all pages and their respective paths.
+fn pages() -> Result<Vec<(PathBuf, String)>> {
+    let template_dir = Path::new(TEMPLATE_DIR);
+    let target_dir = Path::new(OUTPUT_DIR);
+
+    // Set `XDG_DATA_HOME` to a bogus directory so regardless of installed
+    // extensions, none of them are ever documented.
+    env::set_var("XDG_DATA_HOME", "/i/n/v/a/l/i/d");
```

Comment:
I don't like using a directory that actually might exist because if our CLI starts messing with it things might go **horribly** wrong.

/dev/null isn't a directory for example, worst-case scenario someone goes ahead and creates a directory if it didn't explicitly find a directory. Now normally that would just be a bad time nuking a potentially existing file, but if you nuke /dev/null to replace it with a /dev/null directory things are going to be **bad**.

I've considered /dev/null myself, but ultimately there's no **advantage** to using it, we don't actually need a null sink or use a null sink. And the hazards are just unnecessary.

Reply:
Roger that...withdrawn...I should have tested it first anyway...b/c it doesn't work:

```sh
❯ XDG_DATA_HOME=/dev/null poetry run phylum-init
 [*] Getting https://github.com/phylum-dev/cli/releases/download/v3.12.1/phylum-aarch64-apple-darwin.zip file ...Done
 [*] Saving https://github.com/phylum-dev/cli/releases/download/v3.12.1/phylum-aarch64-apple-darwin.zip file to /var/folders/gh/wnf14j7n4q34y2t36hq2jz800000gn/T/tmpdu9b4y5c/phylum-aarch64-apple-darwin.zip ...Done
 [*] Getting https://github.com/phylum-dev/cli/releases/download/v3.12.1/phylum-aarch64-apple-darwin.zip.signature file ...Done
 [*] Saving https://github.com/phylum-dev/cli/releases/download/v3.12.1/phylum-aarch64-apple-darwin.zip.signature file to /var/folders/gh/wnf14j7n4q34y2t36hq2jz800000gn/T/tmpdu9b4y5c/phylum-aarch64-apple-darwin.zip.signature ...Done
 [*] Verifying /var/folders/gh/wnf14j7n4q34y2t36hq2jz800000gn/T/tmpdu9b4y5c/phylum-aarch64-apple-darwin.zip with signature from /var/folders/gh/wnf14j7n4q34y2t36hq2jz800000gn/T/tmpdu9b4y5c/phylum-aarch64-apple-darwin.zip.signature ...SUCCESS

    phylum-cli installer

mkdir: /dev/null: Not a directory
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/Users/maxrake/dev/phylum/phylum-ci/src/phylum/init/cli.py", line 404, in main
    subprocess.run(cmd, check=True, cwd=extracted_dir)
  File "/Users/maxrake/.pyenv/versions/3.11.0/lib/python3.11/subprocess.py", line 569, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['sh', 'install.sh']' returned non-zero exit status 1.
```